modules/stnBidAdapter.js

```diff
@@ -0,0 +1,488 @@
+import {
+  logWarn,
+  logInfo,
+  isArray,
+  isFn,
+  deepAccess,
+  isEmpty,
+  contains,
+  timestamp,
+  triggerPixel,
+  isInteger,
+  getBidIdParameter
+} from '../src/utils.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER, VIDEO} from '../src/mediaTypes.js';
+import {config} from '../src/config.js';
+
+const SUPPORTED_AD_TYPES = [BANNER, VIDEO];
+const BIDDER_CODE = 'stn';
+const ADAPTER_VERSION = '6.0.0';
+const TTL = 360;
+const DEFAULT_CURRENCY = 'USD';
+const SELLER_ENDPOINT = 'https://hb.stngo.com/';
+const MODES = {
+  PRODUCTION: 'hb-multi',
+  TEST: 'hb-multi-test'
+}
+const SUPPORTED_SYNC_METHODS = {
+  IFRAME: 'iframe',
+  PIXEL: 'pixel'
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  version: ADAPTER_VERSION,
+  supportedMediaTypes: SUPPORTED_AD_TYPES,
+  isBidRequestValid: function (bidRequest) {
+    if (!bidRequest.params) {
+      logWarn('no params have been set to STN adapter');
+      return false;
+    }
+
+    if (!bidRequest.params.org) {
+      logWarn('org is a mandatory param for STN adapter');
+      return false;
+    }
+
+    return true;
+  },
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const combinedRequestsObject = {};
+
+    // use data from the first bid, to create the general params for all bids
+    const generalObject = validBidRequests[0];
+    const testMode = generalObject.params.testMode;
+
+    combinedRequestsObject.params = generateGeneralParams(generalObject, bidderRequest);
+    combinedRequestsObject.bids = generateBidsParams(validBidRequests, bidderRequest);
+
+    return {
+      method: 'POST',
+      url: getEndpoint(testMode),
+      data: combinedRequestsObject
+    }
+  },
+  interpretResponse: function ({body}) {
+    const bidResponses = [];
+
+    if (body.bids) {
+      body.bids.forEach(adUnit => {
+        const bidResponse = {
+          requestId: adUnit.requestId,
+          cpm: adUnit.cpm,
+          currency: adUnit.currency || DEFAULT_CURRENCY,
+          width: adUnit.width,
+          height: adUnit.height,
+          ttl: adUnit.ttl || TTL,
+          creativeId: adUnit.requestId,
+          netRevenue: adUnit.netRevenue || true,
+          nurl: adUnit.nurl,
+          mediaType: adUnit.mediaType,
+          meta: {
+            mediaType: adUnit.mediaType
+          }
+        };
+
+        if (adUnit.mediaType === VIDEO) {
+          bidResponse.vastXml = adUnit.vastXml;
+        } else if (adUnit.mediaType === BANNER) {
+          bidResponse.ad = adUnit.ad;
+        }
+
+        if (adUnit.adomain && adUnit.adomain.length) {
+          bidResponse.meta.advertiserDomains = adUnit.adomain;
+        }
+
+        bidResponses.push(bidResponse);
+      });
+    }
+
+    return bidResponses;
+  },
+  getUserSyncs: function (syncOptions, serverResponses) {
+    const syncs = [];
+    for (const response of serverResponses) {
+      if (syncOptions.iframeEnabled && response.body.params.userSyncURL) {
```

Comment:
Update to use `deepAccess` for `response.body.params.userSyncURL` and `response.body.params.userSyncPixels`

Reply:
Thanks for noticing, changed the lines to use `deepAccess` instead